orca-core/src/test/groovy/com/netflix/spinnaker/orca/config/UserConfiguredUrlRestrictionsSpec.groovy

```diff
@@ -30,22 +35,206 @@ class UserConfiguredUrlRestrictionsSpec extends Specification {
         .withAllowedHostnamesRegex("")
-        .withRejectedIps([])
-        .withAllowedSchemes(['https'])
+        .build())
+
+    when:
+    config.validateURI(uri)
+
+    then:
+    thrown(IllegalArgumentException.class)
+
+    where:
+    uri << ['https://www.test.com', 'https://201.152.178.212', 'https://www.test_underscore.com']
+  }
+
+  @Unroll
+  def 'should exclude common internal URL schemes by default'() {
+    given:
+    UserConfiguredUrlRestrictions config = spyOn(new UserConfiguredUrlRestrictions.Builder().build())
+
+    when:
+    config.validateURI(uri)
+
+    then:
+    thrown(IllegalArgumentException.class)
+
+    where:
+    uri << [
+        'https://orca',
+        'https://clouddriver.svc.cluster.local',
+        'https://echo.internal',
+        'https://orca.spinnaker',
+        'https://orca/admin',
+        'https://orca.spinnaker/admin'
+    ]
+  }
+
+  @Unroll
+  def 'should allow non-internal URLs by default'() {
+    given:
+    UserConfiguredUrlRestrictions config = spyOn(new UserConfiguredUrlRestrictions.Builder().build())
+
+    when:
+    URI validatedUri = config.validateURI(uri)
+
+    then:
+    noExceptionThrown()
+    validatedUri
+
+    where:
+    uri << ['https://google.com', 'https://spinnaker.io/foo/bar', 'https://echo.external', 'http://foo.bar']
+  }
+
+  @Unroll
+  def 'should reject localhost by default'() {
+    given:
+    // Note that this does not use the spy, since we want to actually resolve these
+    UserConfiguredUrlRestrictions config = new UserConfiguredUrlRestrictions.Builder().build()
+
+    when:
+    config.validateURI(uri)
+
+    then:
+    thrown(IllegalArgumentException.class)
+
+    where:
+    uri << ['https://localhost', 'http://localhost', 'http://127.0.0.1', 'https://::1']
+  }
+
+  @Unroll
+  def 'should accept localhost when configured, regardless of name filter'() {
+    given:
+    // Note that this does not use the spy, since we want to actually resolve these
+    UserConfiguredUrlRestrictions config = new UserConfiguredUrlRestrictions.Builder()
+        .withAllowedHostnamesRegex("this_definitely_doesnt_match_localhost")
         .withRejectLocalhost(false)
-        .withRejectLinkLocal(false)
         .build()
 
+    when:
+    URI validatedUri = config.validateURI(uri)
+
+    then:
+    noExceptionThrown()
+    validatedUri
+
+    where:
+    uri << ['https://localhost', 'http://localhost', 'http://127.0.0.1', 'https://::1']
+  }
+
+  @Unroll
+  def 'rejects private IP addresses by default'() {
+    given:
+    UserConfiguredUrlRestrictions config = spyOn(new UserConfiguredUrlRestrictions.Builder().build())
+
     when:
     config.validateURI(uri)
 
     then:
     thrown(IllegalArgumentException.class)
 
     where:
-    uri << ['https://www.test.com', 'https://201.152.178.212', 'https://www.test_underscore.com']
+    uri << [
+        'https://192.168.0.1',
+        'http://172.16.0.1',
+        'http://10.0.0.1',
+        'https://fd12:3456:789a:1::1',
```

Comment:
The ipv6 standard is pretty weird, since the `:` is rather overloaded - in any case, the code within can now exclude them in both forms.